packages/grid/_modules_/grid/hooks/features/useApiRef.ts

```diff
@@ -6,8 +6,13 @@ import { EventEmitter } from '../../utils/EventEmitter';
 /**
  * Hook that instantiate an ApiRef to pass in component prop.
  */
+
+function createGridApi(): GridApi {
+  return new EventEmitter() as GridApi;
+}
+
 export function useApiRef(apiRefProp?: ApiRef): ApiRef {
-  const internalApiRef = React.useRef<GridApi>(new EventEmitter() as GridApi);
+  const internalApiRef = React.useRef<GridApi>(createGridApi());
```

Comment:
It's just a thought but maybe sometimes we are overdoing it with the abstractions or maybe it's just me.